### PR TITLE
Allow garden-external-networker to be used on windows

### PR DESCRIPTION
**Pull request by sunjayBhatia**
- The filelock implementation is windows specific
- FileLocker.Open now returns a LockedFile interface. This is so that on
windows the file is guaranteed to be unlocked prior to being closed.

[#146772321]

Signed-off-by: Sam Smith <sesmith177@gmail.com>

### CODE REVIEW

File: src/lib/filelock/filelock_windows.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+)
+
+// Open will open and lock a file.  It blocks until the lock is acquired.
+// If the file does not yet exist, it creates the file, and any missing
+// directories above it in the path.  To release the lock, Close the file.
+func (l *locker) Open() (LockedFile, error) {
+	dir := filepath.Dir(l.path)
+	err := os.MkdirAll(dir, 0700)
+	if err != nil {
+		panic(err)
```
Comment (rosenhouse) [inline src/lib/filelock/filelock_windows.go]: Why panic here?  We try to avoid panics in library code.

**Comment by rosenhouse** [comment]
Not to cause delays, but might it make sense to extract this file locker as a separate library?  I wonder if other projects would benefit from a cross-platform Golang file lock library?

Happy to be talked out of this, just wondering what y'all think.

File: src/lib/filelock/filelock_windows.go (hunk 1 above)
Comment (sunjayBhatia) [inline src/lib/filelock/filelock_windows.go]: @rosenhouse good catch, we will fix it

Comment (sunjayBhatia) [inline src/lib/filelock/filelock_windows.go]: @rosenhouse actually, its in the original code: https://github.com/cloudfoundry-incubator/cf-networking-release/blob/develop/src/lib/filelock/filelock.go#L25

should we remove that as part of this PR?

**Comment by sunjayBhatia** [comment]
@rosenhouse we wouldn't be opposed to that, i think there are a lot of places where people have reimplemented a file locker (guardian, garden-external-networker, silk)

Does it make sense to pull that out as a separate refactor?

File: src/lib/filelock/filelock_windows.go (hunk 1 above)
Comment (rosenhouse) [inline src/lib/filelock/filelock_windows.go]: uh yeah, if you don't mind :-)  Code review for the win!

**Comment by rosenhouse** [comment]
I'm curious what other @cloudfoundry-incubator/cf-container-networking folks think too

File: src/lib/filelock/filelock_windows.go (hunk 1 above)
Comment (sunjayBhatia) [inline src/lib/filelock/filelock_windows.go]: @rosenhouse panics are gone!

**Pull request merged**